---
title: "First R code: MAF and MR calculation"
name: "Jinliang Yang"
date: 10-24-2023
output:
  word_document: default
  html_notebook: default
---

# Path normalization

Note that you have to learn how to use `markdown` for the text chunk

## Normalize path

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
#library(tidyverse)
```


# Data description

In this dataset, we have n=1503 individuals and n=1094 SNPs
The SNP coding here is 0,1,2 and "-" indicates NA

```{r, eval=TRUE}
data <- read.csv("data/ng_rice_snp.csv")
```


## Calculating the missing rate


```{r}
missing_count = rowSums(data[, -1] == "-") #adding all the occurances of "-" per each row.
total_individuals = ncol(data) - 1 #calculating number of individual samples we have.
missing_rate = missing_count / total_individuals

# Create a output data.frame
out <- data[, 1:2]
out$missing_rate = missing_rate #creating a new column missing rate back in sheet.
```

visualize the missing rate results

```{r}
pdf("graphs/missing_rate.pdf", height=5, width=5)
hist(out$missing_rate, breaks=50, col="red", xlab="SNP missing Rate", main="Missing Rate")
dev.off()
```


#calculating MAF

```{r}
#creating a function to calculate MAF
calculate_maf = function(row) {
  #Exclude the marker name and missing data
  genotypes = row[!is.na(row) & row != "-"]
  #Count of each allele (each '1' counts as one of each allele, '0' and '2' count as two of one allele)
  ref_allele = sum(genotypes == 0) * 2 + sum(genotypes == 1)
  alt_allele = sum(genotypes == 2) * 2 + sum(genotypes == 1)
  #Total alleles
  total_alleles = ref_allele + alt_allele
  #Frequencies
  ref_freq = ref_allele / total_alleles
  alt_freq = alt_allele / total_alleles
  return(min(ref_freq, alt_freq)) #Return the minor allele frequency
}
#Appling the function to each row and store the result in the maf
maf = apply(data[, -1], 1, calculate_maf)  # excluding the first column with marker names and looping over each row.
out$maf = maf #updating new column with maf in excell sheet.
#Saving the data frame with the new 'missing_rate' and 'maf' columns to a new CSV file
write.csv(out[,-2], file = "cache/res_mr_maf.csv", row.names = FALSE, quote = FALSE)

```

```{r}
pdf("graphs/maf.pdf", height=5, width=5)
hist(out$maf, breaks=50, col="blue", xlab="MAF", main="Allele freq")
dev.off()
```